server/app/service/package.scala

```diff
@@ -1,23 +1,43 @@
-import shared.model.{TestResult, TestStatus}
+import monix.execution.cancelables.BooleanCancelable
+import shared.model.{Event, TestResult, TestStatus}
+import shared.view.SuiteReportUtil.red
 
 import scala.util.{Failure, Success, Try}
 
 package object service {
   //scalatest keyword
   val testFailedMarker = "FAILED"
 
-  implicit class RichRunnerString(val output: Try[String]) extends AnyVal {
-    def check = output.filter(!_.contains(testFailedMarker))
+  type CheckNext = (String) => Unit
+
+  type PushResult = (Event) => Unit
+
+  type OnBlockComplete = Try[Unit] => Unit
+
+  def testSync: (CheckNext, (=> Unit) => Event) = {
+    val (checkNext, testResultFromTry) = testAsync(identity)
+
+    def getTestResult(block: => Unit) = testResultFromTry(Try(block))
+
+    checkNext -> getTestResult _
   }
 
-  def testResult(report: Try[String]): TestResult =
-    report match {
-      case Success(s) =>
-        val status =
-          if (s.contains(testFailedMarker)) TestStatus.FailedByTest
-          else TestStatus.Passed
-        TestResult(status.toString)
-      case Failure(e) =>
-        TestResult(TestStatus.FailedByCompilation.toString, e.getMessage)
+  def testAsync[A](pushResult: (Event) => A): (CheckNext, Try[Unit] => A) = {
```

Comment:
I prefer to give long name, so that next time I will understand meaning of this method within 5 sec, rather than 5 mins.